src/main/java/net/revelc/code/apilyzer/maven/plugin/AnalyzeMojo.java

```diff
@@ -453,6 +641,16 @@ private boolean checkClass(Class<?> clazz, Set<String> publicSet, PrintStream ou
         continue;
       }
 
+      if (class1.equals(clazz)) {
+        // what? see org.apache.hadoop.fs.Options$CreateOpts source in hadoop 2.7.1
```

Comment:
A better comment might be: "skip this inner class if it is contained within itself; a class can be an inner class of itself when it is declared inside its superclass, because child classes also contain the inner classes of their parents".